h/views/admin/oauthclients.py

```diff
@@ -150,7 +156,12 @@ def _update_appstruct(self):
     def _template_context(self):
         return {"form": self.form.render()}
 
-    @view_config(request_method="POST", request_param="delete")
-    def delete(self):
-        self.request.db.delete(self.client)
-        return HTTPFound(location=self.request.route_url("admin.oauthclients"))
+    @classmethod
+    def _get_client(cls, request):
+        client_id = request.matchdict.get("id")
+
+        try:
+            return request.db.query(AuthClient).filter_by(id=client_id).one()
+        except (NoResultFound, StatementError) as err:
+            # Statement errors happen if the id is invalid
+            raise HTTPNotFound() from err
```

Comment:
I don't think having this in a separate file makes anything easier to test or understand